CMakeLists.txt

```diff
@@ -366,7 +366,15 @@ endif()
 if(CELERITAS_USE_VecGeom)
   set(_min_vecgeom_version 1.2.4)
   if(NOT VecGeom_FOUND)
-    find_package(VecGeom ${_min_vecgeom_version} REQUIRED)
+    # TODO: remove once we require a veccore version including https://github.com/root-project/veccore/commit/743566fac1e9b2eaeb0f0b63242442ba430e0cc0
+    if(POLICY CMP0146 AND CELERITAS_USE_CUDA)
+      cmake_policy(PUSH)
+      cmake_policy(SET CMP0146 OLD)
+      find_package(VecGeom ${_min_vecgeom_version} REQUIRED)
+      cmake_policy(POP)
+    else()
+      find_package(VecGeom ${_min_vecgeom_version} REQUIRED)
+    endif()
```

Comment:
Could this be moved into FindVecgeom.cmake? As it is, would we still get the warning/error from downstream packages using the `find_dependency` call (or is that OK because it doesn't have the newer cmake policy?)

Reply:
We can move this to FindVecgeom.cmake. CMake >= 3.27 won't output any error/warning from downstream packages. CMake versions between 3.10 (included) and 3.27 (not included) don't know the policy so we'd get a warning whenever CMake decided to introduce one but this is also the case without these changes.